src/plugins/data/common/datasets/types.ts

```diff
@@ -88,12 +110,69 @@ export interface DataStructure {
   id: string;
   /** Human-readable name of the data structure */
   title: string;
-  /** The type of the data structure, registered by other classes */
+  /** The type of the data structure */
   type: string;
   /** Optional reference to the parent data structure */
   parent?: DataStructure;
   /** Optional array of child data structures */
   children?: DataStructure[];
+  /** Optional metadata for the data structure */
+  meta?: DataStructureMeta;
```

Comment:
do we think it makes sense to have the dataset just have a datastructure property instead of a datasource property? and we can construct the datasource from the datastructure?

Reply:
The data source type has the connection details as well. Data structure requires 2 objects to show cluster and connection details and makes the reference bigger when you store it in the url

Reply:
Yeah that makes sense. Will keep as is.

I do like the data structure object. Thanks for the clever idea @ashwin-pc. Seems super obvious that data sources and connections are types of data structures and gives us a flexible framework.